libraries/audio/src/AudioHRTF.cpp

```diff
@@ -388,7 +388,12 @@ void crossfade_4x2_AVX2(float* src, float* dst, const float* win, int numFrames)
 void interpolate_AVX2(const float* src0, const float* src1, float* dst, float frac, float gain);
 
 static void FIR_1x4(float* src, float* dst0, float* dst1, float* dst2, float* dst3, float coef[4][HRTF_TAPS], int numFrames) {
+#ifndef STACK_PROTECTOR
+    // Enabling -fstack-protector on gcc causes an undefined reference to FIR_1x4_AVX512 here
     static auto f = cpuSupportsAVX512() ? FIR_1x4_AVX512 : (cpuSupportsAVX2() ? FIR_1x4_AVX2 : FIR_1x4_SSE);
+#else
+    static auto f = cpuSupportsAVX2() ? FIR_1x4_AVX2 : FIR_1x4_SSE;
+#endif
```

Comment:
This part of the code is a bit too much for me.
My understanding is that you are disabling AVX512 support when `-fstack-protector` is not in use (which it isn't). So you are just future proofing in case someone gets `-fstack-protector` to work.

Reply:
No, it's the reverse. Enabling memory debugging enables -fstack-protector-strong, which breaks FIR_1x4_AVX512 for some reason. So this ensures that function isn't used in that case.

The original hifi code just disabled fstack-protector instead, but it could be useful for debugging since it's an additional check.